typespec-tests/src/main/java/com/cadl/visibility/models/RoundTripModel.java

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.cadl.visibility.models;
+
+import com.azure.core.annotation.Immutable;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/** The RoundTripModel model. */
+@Immutable
+public final class RoundTripModel {
+    /*
+     * The name property.
+     */
+    @JsonProperty(value = "name", required = true)
+    private String name;
+
+    /*
+     * The secretName property.
+     */
+    @JsonProperty(value = "secretName")
+    private String secretName;
+
+    /**
+     * Creates an instance of RoundTripModel class.
+     *
+     * @param name the name value to set.
+     * @param secretName the secretName value to set.
+     */
+    @JsonCreator
+    public RoundTripModel(
+            @JsonProperty(value = "name", required = true) String name,
+            @JsonProperty(value = "secretName") String secretName) {
+        this.name = name;
+        this.secretName = secretName;
+    }
```

Comment:
Most change on generated code happens here.

Option `required = true` in annotation is removed from secret property, so that Jackson will not report error when de-serialize from response (which does not contain the secret property).